include/git2/config.h

```diff
@@ -735,6 +750,7 @@ GIT_EXTERN(int) git_config_parse_path(git_buf *out, const char *value);
  * @param regexp regular expression to match against config names (can be NULL)
  * @param callback the function to call on each variable
  * @param payload the data to pass to the callback
+ * @return non-zero to terminate the iteration.
```

Comment:
This comment is really about how the callback behaves, not the function itself.

```suggestion
 * @return 0 or an error code.
```